packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2756,17 +2761,37 @@ export class ContainerRuntime
 				 */
 				messagesWithPendingState = [
 					{
-						message: inboundBatch.keyMessage,
+						message: inboundResult.batchStart.keyMessage,
 						localOpMetadata: undefined,
 					},
 				];
 				// Empty batch message is a non-runtime message as it was generated by the op grouping manager.
 				runtimeBatch = false;
 			}
-			this.processBatch(messagesWithPendingState, local, savedOp, runtimeBatch);
+			if (inboundResult.type !== "fullBatch") {
+				const [nextMessage, ...nothing] = messagesWithPendingState;
+				assert(nothing.length === 0, "Partial batch should have exactly one message");
+				messagesWithPendingState = [nextMessage];
+			}
+
+			const locationInBatch: { batchStart: boolean; batchEnd: boolean } =
+				inboundResult.type === "fullBatch"
+					? { batchStart: true, batchEnd: true }
+					: inboundResult.type === "batchStartingMessage"
```

Comment:
This is difficult to reason about because the type `"batchStartingMessage"` isn't implemented yet. I'd suggest removing this code and the type altogether until it is actually implemented.